src/test/java/org/jvnet/hudson/plugins/port_allocator/PortAllocationWorkflowTest.java

```diff
@@ -0,0 +1,105 @@
+package org.jvnet.hudson.plugins.port_allocator;
+
+import hudson.model.Label;
+import hudson.model.Node;
+import hudson.model.Result;
+import hudson.model.queue.QueueTaskFuture;
+import hudson.slaves.DumbSlave;
+import hudson.slaves.NodeProperty;
+import hudson.slaves.RetentionStrategy;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Before;
+import org.junit.FixMethodOrder;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runners.MethodSorters;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import java.util.Collections;
+import java.util.List;
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class PortAllocationWorkflowTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
```

Comment:
To meaningfully test anything relating to serialization, you will need to use `RestartableJenkinsRule`.

I think you should be looking at the `xvnc` plugin for examples of analogous work.